test/e2e/singlecluster/leaderworkerset_test.go

```diff
@@ -422,7 +422,7 @@ var _ = ginkgo.Describe("LeaderWorkerSet integration", func() {
 				})
 
 				ginkgo.By("Check workload for group 2 is deleted", func() {
-					util.ExpectObjectToBeDeleted(ctx, k8sClient, createdWorkload2, false)
+					util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, createdWorkload2, false, util.LongTimeout)
```

Comment:
Just checking.
Does the root cause is taking more time to delete Workloads?

Reply:
Thank you for clarifying that. In that case, do you have any reason why we do not verify if group 2 worker StatefulSet is removed before this checking?

Reply:
I'm not sure if it's useful. The Pod group combines Pods from both leader and worker StatefulSets. Additionally, we don’t need to test the LWS staff; we need to confirm that the Pods are deleted and the Workload is removed.

Reply:
> In that case, do you have any reason why we do not verify if group 2 worker StatefulSet is removed before this checking?

I would rather not do it, because StatefulSet is rather a technical detail of LWS. Especially, the e2e should show how users and Kueue interact with LWS, they don't need to know about the internal use of STS.